_posts/2025-02-05-#31363.md

```diff
@@ -10,17 +10,263 @@ status: upcoming
+removes mempool transactions that would be inconsistent with chainstate.
+- [Block disconnection](https://github.com/bitcoin/bitcoin/blob/8fa10edcd1706a1f0dc9d8c3adbc8efa3c7755bf/src/validation.cpp#L296-L390)
+(aka reorganization) resubmits the transactions from the previous chain that are no longer in the new chain.
+- [Block assembly](https://github.com/bitcoin/bitcoin/blob/8fa10edcd1706a1f0dc9d8c3adbc8efa3c7755bf/src/node/miner.cpp#L107-L428)
+(aka mining) fills a block template using mempool transactions, optimizing for fee revenue.
+- [Mempool eviction](https://github.com/bitcoin/bitcoin/blob/8fa10edcd1706a1f0dc9d8c3adbc8efa3c7755bf/src/txmempool.cpp#L1158-L1200)
+is an internal process that decides what transactions to delete when the memory usage exceeds configured limits. It is
+triggered in multiple places, including Mempool validation and Block disconnection.
+
+[`CTxMemPoolEntry`](https://github.com/bitcoin/bitcoin/blob/8fa10edcd1706a1f0dc9d8c3adbc8efa3c7755bf/src/kernel/mempool_entry.h#L79-L106)
+stores references to its parents and children and tracks the {count, total vsize, total fees} of its ancestor and
+descendant sets.  Each time a transaction is added or removed, `CTxMemPool` uses the references to traverse ancestors
+and/or descendants recursively, updating the cached ancestor and descendant information.
+
+### TxGraph Design and Usage
+
+While the current `CTxMemPool` class does not contain a data structure dedicated to storing the graph representation
+itself, "the graph already exists" in the form of `CTxMemPoolEntry`'s parent/child references and cached
+ancestor/descendant information.
+
+The responsibility of keeping this graph up-to-date and within its specified limits largely belongs to `CTxMemPool`, but
+occasionally bleeds out to other areas. For example, in mempool validation, `MemPoolAccept` queries `CTxMemPool` for the
+ancestors of a transaction in order to:
+
+- Calculate whether this transaction would exceed the ancestor/descendant limits. However, sometimes, when evaluating a
+  replacement, we apply an [RBF Carve Out](https://github.com/bitcoin/bitcoin/blob/8fa10edcd1706a1f0dc9d8c3adbc8efa3c7755bf/src/validation.cpp#L950-L984),
+literally increasing the limit by 1, to avoid double-counting the replacement and to-be-replaced transactions. Yuck!
+- Check whether a TRUC transaction meets its additional topology restrictions. However we first need to [aggregate the
+  in-package and in-mempool ancestors](https://github.com/bitcoin/bitcoin/blob/8fa10edcd1706a1f0dc9d8c3adbc8efa3c7755bf/src/validation.cpp#L1551-L1558).
+- When evaluating a package, the precise ancestor/descendant counts of each package transaction and connected mempool
+  transactions are not calculated before submission. Instead, the aggregate package count and size are used. This
+approximation avoids complexity, but can result in overestimation for topologies beyond 1-parent-1-child.
```

Comment:
> The responsibility of keeping this graph up-to-date and within its specified limits largely belongs to `CTxMemPool`

Are you talking about the "emergent" graph that exists on master, or the new TxGraph? If the former, perhaps this section might be a bit of information overload - would it make sense to leave it out?